### PR TITLE
Namespace classes

### PR DESCRIPTION
**Pull request by boxadesign**
## BREAKING CHANGE

Namespace all css classes by prefixing `ons-`. This is to prevent class naming conflicts in services that may have their own set of stylesheets.

### Why is this a breaking change?

In any service that implements ONS DS css classes manually into component configurations or templates to control layout e.g a utility class such as `u-mt-s` or a grid layout with `col-9`.  These "hardcoded" classes will need to be updated to include the `ons-` prefix to the class, so `u-mt-s` would need to be changed to `ons-u-mt-s`.

### CODE REVIEW

**Comment by jrbarnes9** [comment]
[Namespace styles across DS](https://trello.com/c/GrgWL7Xn/1209-namespace-styles-across-ds)

**Review by jrbarnes9 — COMMENTED** [review]
Sub navigation anchor links are missing their dashed prefix.  (assuming this is inherited from the rebuild PR, but I can't check that because the netlify build is failing)
Screenshot 2021-08-20 at 15 57 10

**Comment by jrbarnes9** [comment]
Broken link
Screenshot 2021-08-20 at 16 03 08

**Review by jrbarnes9 — CHANGES_REQUESTED** [review]
- The classes in the Typography type scale table in need updating.
- In `src/components/input/_macro.njk` line 129 ` js-input-legend` missing prefix
- In `src/components/field/_field.scss` line 16, `field__item--w-{x}` missing prefix
- In `src/components/input/_macro.njk` line 97 and `src/components/search/_macro.njk` line 13, there's a typo `attribues` and `params.searchButton.attribues`
- In `src/components/download-resources/download-resources.js`: 
    - lines 1020 & 1064, `adv-filter__panel...` are missing prefix,
    - lines 550, 553, 647, 690, 738 & 741 `u-hidden` missing prefix
    - line 1064 `adv-filter__...` missing prefix
    - line 1054 `no-scroll` missing prefix
    - lines 320/321 `js-filter___...` missing prefix
- In `src/components/accordion/examples/accordion/index.njk` hardcoded classes in accordion `content` params are missing prefixes
- In all `onsButton` examples` the `classes` params are missing prefix
- Lots of component examples the use single quotes or the `classes` param value are missing prefix

**Comment by jrbarnes9** [comment]
- In `src/components/question/examples/question-ccs/index.njk`, line 25, `highlight` class missing prefix
- In `src/components/question/examples/question-interviewer-note/index.njk`, line 25, `instruction` class missing prefix

**Review by jrbarnes9 — COMMENTED** [review]
- In `src/components/tabs/tabs.js` line 143, `.tab...` class missing prefix

**Review by jrbarnes9 — CHANGES_REQUESTED** [review]
Just headings example in typography still missing the prefixes
Screenshot 2021-09-02 at 09 39 45

**Review by rmccar — COMMENTED** [review]
image
The Facebook icon might need looking at

**Review by rmccar — COMMENTED** [review]
image
`ons-` missing from these classes

**Review by rmccar — COMMENTED** [review]
Also on the grid page some classes listed start with full stops and some don't. Should probably have all or nothing

**Review by rmccar — COMMENTED** [review]
image
Arabic example no longer right aligned

**Review by rmccar — CHANGES_REQUESTED** [review]
image
Facebook icon needs looking at

**Comment by rmccar** [comment]
> image
> Facebook icon needs looking at

Accidentally added this twice

**Comment by rmccar** [comment]
image
Needs `.ons-`

**Comment by rmccar** [comment]
image
Collapsible accessibility docs need `ons-`

**Comment by rmccar** [comment]
image
Either `.ons-` needs to be added or docs need updating to say to add a description param not a class. I think the latter is better

**Comment by rmccar** [comment]
image
Current page in pagination is no longer highlighted

**Comment by rmccar** [comment]
image
Warning panel spacing is off

**Comment by rmccar** [comment]
Status component needs `ons-` added to all classes

**Comment by rmccar** [comment]
image
`.ons-` needed to be added to summary docs

**Comment by jrbarnes9** [comment]
image


<details><summary>Browser metadata</summary>





```
Path:      /components/radios/
Browser:   Chrome 92.0.4515.159 on Mac OS 10.14.6
Viewport:  1331 x 794 @1x
Language:  en
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-1638--ons-design-system.netlify.app%2Fcomponents%2Fradios%2F&os=OS%20X&os_version=Mojave&browser=Chrome&browser_version=92.0&resolution=1331x794&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-1638--ons-design-system.netlify.app/components/radios/) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=6138b7e25c0bce00070c04e4&commentID=6138bb8c73233dc8187649f2&resolution=resolved)

**Comment by jrbarnes9** [comment]
[uploading...]


<details><summary>Browser metadata</summary>





```
Path:      /components/checkboxes/
Browser:   Chrome 92.0.4515.159 on Mac OS 10.14.6
Viewport:  1331 x 794 @1x
Language:  en
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-1638--ons-design-system.netlify.app%2Fcomponents%2Fcheckboxes%2F&os=OS%20X&os_version=Mojave&browser=Chrome&browser_version=92.0&resolution=1331x794&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-1638--ons-design-system.netlify.app/components/checkboxes/) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=6138b7e25c0bce00070c04e4&commentID=6138bbdbf0b403c596160abf&resolution=resolved)

**Comment by jrbarnes9** [comment]
image


<details><summary>Browser metadata</summary>





```
Path:      /components/checkboxes/
Browser:   Chrome 92.0.4515.159 on Mac OS 10.14.6
Viewport:  1331 x 794 @1x
Language:  en
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-1638--ons-design-system.netlify.app%2Fcomponents%2Fcheckboxes%2F&os=OS%20X&os_version=Mojave&browser=Chrome&browser_version=92.0&resolution=1331x794&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-1638--ons-design-system.netlify.app/components/checkboxes/) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=6138b7e25c0bce00070c04e4&commentID=6138bbf3048bf9c7e6dbc931&resolution=resolved)

File: src/tests/spec/cookies/cookies-banner.spec.js [hunk 1]
```diff
@@ -102,52 +102,52 @@ describe('Component: Cookie banner', function() {
...
+    '<input type="radio" class="ons-js-radio" id="settings-off" name="cookies-settings" value="off">' +
+    '<input type="radio" class="ons-js-radio" name="cookies-usage" value="on">' +
+    '<input type="radio" class="ons-js-radio" name="cookies-usage" value="off">' +
+    '<input type="radio" class="ons-js-radio" name="cookies-campaigns" value="on">' +
+    '<input type="radio" class="ons-js-radio" name="cookies-campaigns" value="off">' +
     '<button id="submit-button" type="submit">Submit</button>' +
     '</form>' +
-    '<div class="cookies-confirmation-message u-d-no">' +
-    '<a class="js-return-link" href="#">Return to previous page</a>' +
+    '<div class="ons-cookies-confirmation-messageons-u-d-no">' +
```
Comment (jrbarnes9) [inline src/tests/spec/cookies/cookies-banner.spec.js:142]:
```suggestion
    '<div class="ons-cookies-confirmation-message ons-u-d-no">' +
```

File: src/components/panel/_macro.njk [hunk 2]
```diff
@@ -58,24 +58,24 @@
                         {% set defaultTitleTag = "div" %}
                     {% endif %}
                     {% set titleTag = params.titleTag | default(defaultTitleTag) %}
-                    <div class="panel__header">
-                        <{{ titleTag }} id="error-summary-title" {% if params is defined and params and params.type is defined and params.type %}data-qa="{{ params.type }}-header"{% endif %} class="panel__title u-fs-r--b">{{ params.title | safe }}</{{ titleTag }}>
+                    <div class="ons-panel__header">
+                        <{{ titleTag }} id="error-summary-title" {% if params is defined and params and params.type is defined and params.type %}data-qa="{{ params.type }}-header"{% endif %} class="ons-panel__titleons-u-fs-r--b">{{ params.title | safe }}</{{ titleTag }}>
```
Comment (jrbarnes9) [inline src/components/panel/_macro.njk:62]:
```suggestion
                        <{{ titleTag }} id="error-summary-title" {% if params is defined and params and params.type is defined and params.type %}data-qa="{{ params.type }}-header"{% endif %} class="ons-panel__title ons-u-fs-r--b">{{ params.title | safe }}</{{ titleTag }}>
```

File: src/scss/utilities/_pad.scss [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// e.g u-pl-m
+// e.gons-u-pl-m
```
Comment (jrbarnes9) [inline src/scss/utilities/_pad.scss:1]:
```suggestion
// e.g ons-u-pl-m
```

File: src/pages/patterns/news/examples/article/index.njk [hunk 4]
```diff
@@ -72,7 +72,7 @@ layout: none
 
             <p>For more information, including how to find a local census support centre, please visit <a href="#0">www.census.gov.uk</a>.</p>
 
-            <div class="u-mt-l u-mt-xl@m">
+            <div class="ons-u-mt-lons-u-mt-xl@m">
```
Comment (jrbarnes9) [inline src/pages/patterns/news/examples/article/index.njk:75]:
```suggestion
            <div class="ons-u-mt-l ons-u-mt-xl@m">
```

**Review by jrbarnes9 — CHANGES_REQUESTED** [review]

File: src/components/button/index.njk [hunk 5]
```diff
@@ -71,7 +71,7 @@ Setting the `buttonStyle` param to `exit` adds the appropriate icon for a 'sign
 }}
 
 ### Disabled
-Use `btn--disabled` with the `disabled=true` element attribute for buttons that have no event/action but need to be displayed.
+Use `ons-btn--disabled` with the `disabled=true` element attribute for buttons that have no event/action but need to be displayed.
```
Comment (rmccar) [inline src/components/button/index.njk:74]:
```suggestion
Use `.ons-btn--disabled` with the `disabled=true` element attribute for buttons that have no event/action but need to be displayed.
```

File: src/components/summary/index.njk [hunk 6]
```diff
@@ -45,7 +45,7 @@ The summary component is made accessible by using the following `aria` attribute
             {
                 "tds": [
                     {
-                        "value": "<code>summary__button</code>"
+                        "value": "<code>ons-summary__button</code>"
```
Comment (rmccar) [inline src/components/summary/index.njk:48]:
```suggestion
                        "value": "<code>.ons-summary__button</code>"
```

**Review by rmccar — CHANGES_REQUESTED** [review]

**Comment by rmccar** [comment]
image
Underline missing from skip to link

**Comment by boxadesign** [comment]
> image
> Underline missing from skip to link

Won't do on this branch

**Review by jrbarnes9 — APPROVED** [review]

**Pull request merged**